src/components/forms/AddressInput/index.tsx

```diff
@@ -39,50 +41,65 @@ const AddressInput = ({
   validators = [],
   defaultValue,
   disabled,
-}: AddressInputProps): React.ReactElement => (
-  <>
-    <Field
-      className={className}
-      component={TextField as any}
-      defaultValue={defaultValue}
-      disabled={disabled}
-      inputAdornment={inputAdornment}
-      name={name}
-      placeholder={placeholder}
-      testId={testId}
-      text={text}
-      type="text"
-      spellCheck={false}
-      validate={composeValidators(required, mustBeEthereumAddress, ...validators)}
-    />
-    <OnChange name={name}>
-      {async (value: string) => {
-        const address = trimSpaces(value)
-        // A crypto domain name
-        if (isValidEnsName(address) || isValidCryptoDomainName(address)) {
-          try {
-            const resolverAddr = await getAddressFromDomain(address)
-            const formattedAddress = checksumAddress(resolverAddr)
-            fieldMutator(formattedAddress)
-          } catch (err) {
-            logError(Errors._101, err.message)
-          }
-        } else {
-          // A regular address hash
-          let checkedAddress = address
-          // Automatically checksum valid (either already checksummed, or lowercase addresses)
-          if (isValidAddress(address)) {
+}: AddressInputProps): ReactElement => {
+  const [isVerifying, setIsVerifying] = useState<boolean>(false)
+  const adornment = isVerifying
+    ? {
+        endAdornment: (
+          <InputAdornment position="end">
+            <CircularProgress size="16px" />
+          </InputAdornment>
+        ),
+      }
+    : inputAdornment
+  return (
+    <>
+      <Field
+        className={className}
+        component={TextField as any}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        readOnly={isVerifying}
```

Comment:
Not sure it's nice to freeze the UI like this.
It starts verifying only once something domain-looking is entered, right?

Reply:
Yes, but if someone adds any further text then it will then be an incorrect domain, i.e. I type 'iamacook.eth' but then add an extra 'h', it will invalidate the address.

Reply:
Are there any drawbacks if you make it non-readonly?

Reply:
The address will be re-resolved if the address was manipulated during resolution.

Reply:
Oh yeah, I remember fighting with that when I was making this UI component. I wonder if the [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) can come in handy.